subsys/net/ip/connection.c

```diff
@@ -679,13 +699,13 @@ enum net_verdict net_conn_packet_input(struct net_pkt *pkt, uint16_t proto)
 
 		/* Apply protocol-specific matching criteria... */
 
-		if (!(conn->flags & NET_CONN_LOCAL_ADDR_SET)) {
-			continue;
-		}
-
-		local = (struct sockaddr_ll *)&conn->local_addr;
-		if (local->sll_ifindex != net_if_get_by_iface(net_pkt_iface(pkt))) {
-			continue;
+		/* Unbound sockets should collect packets from all interfaces. */
+		if (conn->flags & NET_CONN_LOCAL_ADDR_SET) {
```

Comment:
Strictly speaking this falls under the [required Misra 14.4 rule](https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html).

```suggestion
		if ((conn->flags & NET_CONN_LOCAL_ADDR_SET) != 0U) {
```